hvac/aws_utils.py

```diff
@@ -0,0 +1,42 @@
+import hmac
+from datetime import datetime
+from hashlib import sha256
+
+
+class SigV4Auth(object):
```

Comment:
This probably falls under personal preference, but I was hoping for this to go with a functional set up instead of a Class-based one. If you're amenable to it, I pulled down your branch and added some commits illustrating what I had in mind that could be merged in to your branch: https://github.com/ltm/hvac/compare/auth-aws...jeffwecan:auth-aws?expand=1

Since its not a deal breaker either way, just let me know if you'd rather stick with things as-is and we can get this merged.

Reply:
The `SigV4Auth` approach is (very) loosely based on botocore (i.e. https://github.com/boto/botocore/blob/master/botocore/auth.py#L151). I do think it makes sense to keep the signature algorithm separate from the creation of the `GetCallerIdentity` request since this is specific to Vault. That said I don't have a strong preference either way so I'll defer to you.

Reply:
Make sense, I can live with this as our initial effort given that context.